service/history/transferQueueFactory.go

```diff
@@ -70,25 +68,20 @@ func NewTransferQueueFactory(
 ) queues.Factory {
 	var scheduler queues.Scheduler
 	if params.Config.TransferProcessorEnablePriorityTaskScheduler() {
-		scheduler = queues.NewScheduler(
+		scheduler = queues.NewNamespacePriorityScheduler(
```

Comment:
Why is this NewNamespacePriorityScheduler only used for transfer queue, but not timer or visibility queue?

Reply:
It is used for timer and visibility queue.
Timer: https://github.com/temporalio/temporal/pull/3252/files#diff-53c6b294734809761d2454709c49c906ceaf603223a3bf715b928c9726ba99bfR68
Visibility: https://github.com/temporalio/temporal/pull/3252/files#diff-fb8866471d25cf84b23f1bfb6595143f5e269e6a4efbdeba21d6b211e100d096R61

Reply:
Synced offline. Shard scheduler is using fifo scheduler, host level scheduler is using NamespacePriorityScheduler. Renamed the function and variables to make it easier to read. 

Fifo scheduler and logic for creating/manager shard level scheduler/worker pool can be removed in the future (1.19?).